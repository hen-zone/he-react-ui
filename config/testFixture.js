import createTestContext from 'react-cosmos-test/enzyme';

export default function testFixture(fixture, variant) {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component;
  mount();
  test(`<${displayName ||
    name} /> rendered correctly with ${variant} fixture`, () =>
    expect(getWrapper()).toMatchSnapshot());
}
