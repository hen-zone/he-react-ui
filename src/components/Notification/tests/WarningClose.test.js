import fixture from '../fixtures/WarningClose.fixture.js'
import testFixture from '../../../../config/testFixture'

testFixture(fixture, 'WarningClose')
