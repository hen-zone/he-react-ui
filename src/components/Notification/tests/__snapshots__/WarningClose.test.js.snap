// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> rendered correctly with WarningClose fixture 1`] = `
<Notification
  canClose={true}
  closed={false}
  onClose={[Function]}
  type="warning"
>
  <div
    className="notification warning"
  >
    <Alert
      className="icon warning"
      fill="currentColor"
      height={32}
      width={32}
    >
      <svg
        className="icon warning"
        fill="currentColor"
        height={32}
        viewBox="0 0 20 20"
        width={32}
      >
        <path
          d="M10.882 3.654l6.334 11.875A1 1 0 0 1 16.333 17H3.667a1 1 0 0 1-.883-1.47L9.118 3.653a1 1 0 0 1 1.764 0zM9.25 8v2.5a.75.75 0 0 0 1.5 0V8a.75.75 0 0 0-1.5 0zm.75 7a1 1 0 1 0 0-2 1 1 0 0 0 0 2z"
        />
      </svg>
    </Alert>
    <div
      className="message"
    >
      Warning text. Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur.
    </div>
    <Cross
      className="closeIcon warning"
      fill="currentColor"
      height={32}
      onClick={[Function]}
      width={32}
    >
      <svg
        className="closeIcon warning"
        fill="currentColor"
        height={32}
        onClick={[Function]}
        viewBox="0 0 24 24"
        width={32}
      >
        <path
          d="M12 10.586l3.293-3.293a1 1 0 1 1 1.414 1.414L13.414 12l3.293 3.293a1 1 0 1 1-1.414 1.414L12 13.414l-3.293 3.293a1 1 0 1 1-1.414-1.414L10.586 12 7.293 8.707a1 1 0 1 1 1.414-1.414L12 10.586z"
        />
      </svg>
    </Cross>
  </div>
</Notification>
`;
